src/LightInject/LightInject.cs

```diff
@@ -6537,8 +6543,17 @@ private void BuildImplementationMap(Type implementingType, IServiceRegistry serv
         private void RegisterInternal(Type serviceType, Type implementingType, IServiceRegistry serviceRegistry, ILifetime lifetime)
         {
             var serviceTypeInfo = serviceType.GetTypeInfo();
+
+            
             if (serviceTypeInfo.IsGenericType && serviceTypeInfo.ContainsGenericParameters)
             {
+                var implementingTypeInfo = implementingType.GetTypeInfo();
+
+                if (implementingTypeInfo.IsGenericType && implementingTypeInfo.ContainsGenericParameters && implementingTypeInfo.GenericTypeParameters.Length > serviceTypeInfo.GenericTypeArguments.Length)
```

Comment:
I am afraid that this can be a little too simple. Consider this implementation.
```

    public interface IBar<T1, T2>
    {
    }


    public class Bar<T1, T2> : IBar<string, int>
    {
    }
```

The number of arguments arge correct but they does not close the generic parameters of the implementing type.